events.go

```diff
@@ -133,6 +133,9 @@ func (s *Session) handleNodeEvent(frames []frame) {
 		// TODO: can we be sure the order of events in the buffer is correct?
 		switch f := frame.(type) {
 		case *topologyChangeEventFrame:
+			host, port := s.cfg.translateAddress(f.host.String(), f.port)
```

Comment:
One issue I can see with this is that if the translator is not deterministic then we are going to end up with orphaned connections or nodes in the ring, which will be bad. A few places index on a host's address, it might be worthwhile adding another field to `HostInfo` which stores the translated address and still keeps the address that Cassandra thinks a node is?